core/shared/platform/common/posix/posix_socket.c

```diff
@@ -351,6 +353,306 @@ os_socket_convert_sockaddr(struct sockaddr *addr, uint8_t *buf, size_t buflen,
+{
+    assert(bufsiz);
+
+    socklen_t bufsiz_len = sizeof(bufsiz);
+    if (getsockopt(socket, SOL_SOCKET, SO_SNDBUF, bufsiz, &bufsiz_len) != 0) {
+        return BHT_ERROR;
+    }
+
+    return BHT_OK;
+}
+
+int
+os_socket_set_recv_buf_size(bh_socket_t socket, size_t bufsiz)
+{
+    if (setsockopt(socket, SOL_SOCKET, SO_RCVBUF, &bufsiz, sizeof(bufsiz))
+        != 0) {
+        return BHT_ERROR;
+    }
+
+    return BHT_OK;
+}
+
+int
+os_socket_get_recv_buf_size(bh_socket_t socket, size_t *bufsiz)
+{
+    assert(bufsiz);
+
+    socklen_t bufsiz_len = sizeof(bufsiz);
+    if (getsockopt(socket, SOL_SOCKET, SO_RCVBUF, bufsiz, &bufsiz_len) != 0) {
+        return BHT_ERROR;
+    }
+
+    return BHT_OK;
+}
+
+int
+os_socket_set_keep_alive(bh_socket_t socket, uint8 enable)
+{
+    return os_socket_setbooloption(socket, SOL_SOCKET, SO_KEEPALIVE, enable);
+}
+
+int
+os_socket_get_keep_alive(bh_socket_t socket, uint8 *enabled)
+{
+    return os_socket_getbooloption(socket, SOL_SOCKET, SO_KEEPALIVE, enabled);
+}
+
+int
+os_socket_setsndtimeo(bh_socket_t socket, uint64 timeout_us)
+{
+    struct timeval tv;
+    tv.tv_sec = timeout_us / 1000000UL;
+    tv.tv_usec = timeout_us % 1000000UL;
+    if (setsockopt(socket, SOL_SOCKET, SO_SNDTIMEO, &tv, sizeof(tv)) != 0) {
+        return BHT_ERROR;
+    }
+
+    return BHT_OK;
+}
+
+int
+os_socket_getsndtimeo(bh_socket_t socket, uint64 *timeout_us)
+{
+    struct timeval tv;
+    socklen_t tv_len = sizeof(tv);
+    if (getsockopt(socket, SOL_SOCKET, SO_SNDTIMEO, &tv, &tv_len) != 0) {
+        return BHT_ERROR;
+    }
+    *timeout_us = (tv.tv_sec * 1000000UL) + tv.tv_usec;
+    return BHT_OK;
+}
+
+int
+os_socket_setrecvtimeo(bh_socket_t socket, uint64 timeout_us)
+{
+    struct timeval tv;
+    tv.tv_sec = timeout_us / 1000000UL;
+    tv.tv_usec = timeout_us % 1000000UL;
+    if (setsockopt(socket, SOL_SOCKET, SO_RCVTIMEO, &tv, sizeof(tv)) != 0) {
+        return BHT_ERROR;
+    }
+
+    return BHT_OK;
+}
+
+int
+os_socket_getrecvtimeo(bh_socket_t socket, uint64 *timeout_us)
+{
+    struct timeval tv;
+    socklen_t tv_len = sizeof(tv);
+    if (getsockopt(socket, SOL_SOCKET, SO_RCVTIMEO, &tv, &tv_len) != 0) {
+        return BHT_ERROR;
+    }
+    *timeout_us = (tv.tv_sec * 1000000UL) + tv.tv_usec;
+    return BHT_OK;
+}
+
+int
+os_socket_set_reuse_addr(bh_socket_t socket, uint8 enable)
+{
+    return os_socket_setbooloption(socket, SOL_SOCKET, SO_REUSEADDR, enable);
+}
+
+int
+os_socket_get_reuse_addr(bh_socket_t socket, uint8 *enabled)
+{
+    return os_socket_getbooloption(socket, SOL_SOCKET, SO_REUSEADDR, enabled);
+}
+
+int
+os_socket_set_reuse_port(bh_socket_t socket, uint8 enable)
+{
+    return os_socket_setbooloption(socket, SOL_SOCKET, SO_REUSEPORT, enable);
+}
+
+int
+os_socket_get_reuse_port(bh_socket_t socket, uint8 *enabled)
+{
+    return os_socket_getbooloption(socket, SOL_SOCKET, SO_REUSEPORT, enabled);
+}
+
+int
+os_socket_set_linger(bh_socket_t socket, bh_linger *bh_linger_opts)
```

Comment:
I don't see this (and few other functions here) declared in `platform_api_extension.h` Is that intentional?